src/components/manager/access/GrantForm.tsx

```diff
@@ -0,0 +1,514 @@
+import { Fragment, ReactNode, useCallback, useEffect, useMemo, useState } from "react";
+
+import { Checkbox, Form, Input, Popover, Radio, Select, Space, Spin } from "antd";
+import type { FormInstance, RadioGroupProps, RadioChangeEvent, SelectProps } from "antd";
+
+import { RESOURCE_EVERYTHING, useOpenIdConfig } from "bento-auth-js";
+
+import MonospaceText from "@/components/common/MonospaceText";
+import { useAllPermissions, useGroups } from "@/modules/authz/hooks";
+import {
+    Grant,
+    GrantSubject,
+    PermissionDefinition,
+    Resource as GrantResource,
+    StoredGroup,
+} from "@/modules/authz/types";
+import { useDataTypes } from "@/modules/services/hooks";
+import type { BentoServiceDataType } from "@/modules/services/types";
+import { useProjects } from "@/modules/metadata/hooks";
+import type { Dataset, Project } from "@/modules/metadata/types";
+
+import ExpiryInput from "./ExpiryInput";
+import Resource from "./Resource";
+import Subject from "./Subject";
+import type { InputChangeEventHandler } from "./types";
+
+
+const SUBJECT_EVERYONE: GrantSubject = { everyone: true };
+
+type SubjectInputProps = {
+    value?: GrantSubject;
+    onChange?: (v: GrantSubject) => void;
+};
+
+type SubjectType = "everyone" | "iss-sub" | "iss-client" | "group";
```

Comment:
Literals are repeated a bunch, putting them in consts would help us avoid typos and ease refactors in the future.